server/connectors/bitbucket_connector/api_client.py

```
@@ -15,12 +15,25 @@
+    # Reject path traversal. Decode each segment to a fixed point so that
+    # doubly-encoded forms like ``%252e%252e`` (which downstream clients may
+    # re-decode into ``..``) are also rejected.
+    for segment in parts.path.split("/"):
+        decoded = segment
+        for _ in range(3):
+            previous = decoded
+            decoded = unquote(decoded)
+            if decoded == previous:
+                break
+        if decoded == "..":
+            raise ValueError("URL path contains a traversal segment")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)